src/main/frontend/components/views.cljs

```diff
@@ -1404,9 +1405,136 @@
+         :size :sm
+         :class (str "text-sm px-0 py-0 h-6 " (when-not current-view? "text-muted-foreground"))
+         :on-click (fn [e]
+                     (if (and current-view? (not= (:db/id view) (:db/id view-parent)))
+                       (shui/popup-show!
+                        (.-target e)
+                        (fn []
+                          [:<>
+                           (shui/dropdown-menu-sub
+                            (shui/dropdown-menu-sub-trigger
+                             "Rename")
+                            (shui/dropdown-menu-sub-content
+                             (when-let [block-container-cp (state/get-component :block/container)]
+                               (block-container-cp {} view))))
+                           (shui/dropdown-menu-item
+                            {:key "Delete"
+                             :on-click (fn []
+                                         (p/do!
+                                          (editor-handler/delete-block-aux! view)
+                                          (let [views' (remove (fn [v] (= (:db/id v) (:db/id view))) views)]
+                                            (set-views! views')
+                                            (set-view-entity! (first views'))
+                                            (shui/popup-hide!))))}
+                            "Delete")])
+                        {:as-dropdown? true
+                         :align "start"
+                         :content-props {:onClick shui/popup-hide!}})
+                       (set-view-entity! view)))}
+        (let [display-type (or (:db/ident (get view :logseq.property.view/type))
+                               :logseq.property.view/type.table)]
+          (when-let [icon (:logseq.property/icon (db/entity display-type))]
+            (icon-component/icon icon {:color? true
+                                       :size 15})))
+        (let [title (:block/title view)]
+          (if (= title "")
+            "New view"
+            title)))))
+   (shui/button
+    {:variant :text
+     :size :sm
+     :title "Add new view"
+     :class (str "!px-1 -ml-1 text-muted-foreground hover:text-foreground transition-opacity ease-in duration-300 "
+                 (if hover? "opacity-100" "opacity-75"))
+     :on-click (fn []
+                 (p/let [view (create-view! view-parent view-identity)]
+                   (set-views! (concat views [view]))))}
+    (ui/icon "plus" {:size 15}))])
+
+(rum/defc view-head < rum/static
+  [view-parent view-entity table columns input sorting
+   set-input! add-new-object!
+   {:keys [view-identity title-key additional-actions]
+    :as option}]
+  (let [[hover? set-hover?] (hooks/use-state nil)]
+    [:div.flex.flex-1.flex-wrap.items-center.justify-between.gap-1
+     {:on-mouse-over #(set-hover? true)
+      :on-mouse-out #(set-hover? false)}
+     [:div.flex.flex-row.items-center.gap-2
+      (if (= view-identity :query-result)
```

Comment:
Could we enable multiple views for queries or is that more work?

Reply:
Yes, It's more work, I'll find some time to enable it in another PR.